generic/png.c

```diff
@@ -10,10 +10,30 @@
  *
  * Clement: modified for Torch7.
  */
+#include <assert.h>
+
+/*
+ * Bookkeeping struct for reading png data from memory
+ */
+typedef struct {
+  unsigned char* buffer;
+  png_size_t offset;
+  png_size_t length;
+} libpng_(inmem_buffer);
```

Comment:
just FYI, this does not have to be macro-templated, as it does not have any templated values. It is harmless to template it though.